### PR TITLE
refactor: refactor sdk 1.0.0 api

### PR DESCRIPTION
**Pull request by mrrtree**
# Which issue does this PR close?

Closes #

# Rationale for this change
 
 
SDK redesign unified API style

# What changes are included in this PR?

refactor api

# Are there any user-facing changes?

1. Write use new model `Point`, and query use new model `Row` 
2. Rename `metric` to `table` 
3. Add RouteMode `Direct` and `Proxy`

# How does this change test

all unit test is pass
example is paas

### CODE REVIEW

File: utils/query.go [hunk 1]
```diff
@@ -3,35 +3,202 @@
...
+		rows[rowIdx] = types.Row{
+			Values: make(map[string]types.Value, record.NumCols()),
+		}
+	}
+
+	for colIdx, field := range schema.Fields() {
+		column := record.Column(colIdx)
+		switch column.DataType().ID() {
+		case arrow.STRING:
+			colString, _ := column.(*array.String)
```
Comment (jiacai2050) [inline utils/query.go]:
```
t := value.(typeName)
```
This is enough if you don't want to do type check

File: types/code.go [hunk 2]
```diff
@@ -8,15 +8,14 @@ import (
 )
 
 var (
-	ErrBuiltBuilder        = errors.New("builder has been built, use new one or reset")
-	ErrRowEmptyMetric      = errors.New("builder has been built, use new one or reset")
-	ErrRowEmptyTimestamp   = errors.New("timestamp shoud not be empty")
-	ErrRowEmptyFields      = errors.New("fields should not be empty")
-	ErrRowInvalidFieldType = errors.New("filed invalid type")
+	ErrPointEmptyTable     = errors.New("point table should not be empty")
+	ErrPointEmptyTimestamp = errors.New("point timestamp should not be empty")
```
Comment (jiacai2050) [inline types/code.go]: should large than 0

File: types/route.go [hunk 3]
```diff
@@ -2,8 +2,15 @@
 
 package types
 
+type RouteMode string
+
+const (
```
Comment (jiacai2050) [inline types/route.go:7]: How about this style?
```
type RouteMode int

const (
        Direct RouteMode = iota
        Proxy
)
```

**Review by jiacai2050 — CHANGES_REQUESTED** [review]

File: utils/query.go (hunk 1 above)
Comment (mrrtree) [reply utils/query.go]: ok, fixed

File: types/code.go (hunk 2 above)
Comment (mrrtree) [reply types/code.go]: rename to
```go
	ErrPointEmptyTable     = errors.New("point's table is not set")
	ErrPointEmptyTimestamp = errors.New("point's timestamp is not set")
```

File: types/route.go (hunk 3 above)
Comment (mrrtree) [reply types/route.go:7]: fixed

File: types/code.go (hunk 2 above)
Comment (jiacai2050) [reply types/code.go]: 👍

File: examples/read_write.go [hunk 4]
```diff
@@ -70,22 +70,17 @@ func dropTable(client ceresdb.Client) error {
...
-		SetTimestamp(utils.CurrentMS()).
-		AddTag("name", types.NewStringValue("test_tag2")).
-		AddField("value", types.NewDoubleValue(0.2414)).
-		BuildAndContinue().
-		Build()
-	if err != nil {
-		fmt.Printf("write table build row fail, err: %v\n", err)
-		return err
+	points := make([]types.Point, 0, 2)
+	for i := 0; i < 2; i++ {
```
Comment (jiacai2050) [inline examples/read_write.go:74]: 👍

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Pull request merged**